experiment/prepare.py

```diff
@@ -15,18 +15,81 @@ def __init__(
         rng: np.random.Generator,
         default_params: Path,
         experiment_params: Path,
+        n_samples: int
+        
     ):
         self.n_iterations = n_iterations
         self.rng = rng
         self.default_params = self.load_json(default_params)
         self.experiment_params = self.load_json(experiment_params)
+        self.n_samples = n_samples
 
         # self.covariate_values = self.draw_covariate_values()
 
     def load_json(self, path: Path):
         with open(path) as f:
             return json.load(f)
 
+    def make_scenario_df(self):
+
+        starting_dose_dist = self.rng.triangular(
```

Comment:
If you feel like it, you could use a loop here as in `draw_covariate_values` to reduce repetition.